presto-testing/src/main/java/io/prestosql/testing/datatype/DataTypeTest.java

```diff
@@ -31,16 +33,31 @@
 {
     private final List<Input<?>> inputs = new ArrayList<>();
 
-    private DataTypeTest() {}
+    private boolean runSelectWithWhere;
```

Comment:
Why not always?

Reply:
I was afraid of breaking existing tests.

Reply:
Can we give it a try?

Reply:
Enabling it by default causes problems with:

- TestMySqlTypeMapping.testBasicTypes:107 » NoSuchElement
- TestMySqlTypeMapping.testJson:449 » NoSuchElement
- TestPhoenixSqlTypeMapping.testArray:260 » Runtime line 1:88: Function 'to_date' not registered
- TestPhoenixSqlTypeMapping.testArrayNulls:292 » Runtime ARRAY comparison not supported for arrays with null elements
- TestPhoenixSqlTypeMapping.testDate:239 » Runtime line 1:76: Function 'to_date' not registered
- TestPhoenixSqlTypeMapping.testDecimal:171 » Runtime line 1:1: Invalid numeric literal: 27182818284590452353602874713526624977
- TestPhoenixSqlTypeMapping.testVarbinary:111 » Runtime line 1:81: Function 'decode' not registered

Because of this, tests for the following modules were skipped:
- presto-postgresql
- presto-sqlserver
- presto-iceberg

Reply:
> TestMySqlTypeMapping.testBasicTypes:107 » NoSuchElement

That also shows other problem. When the test fails, there is no hint what went wrong. 
We could eg  issue separate select for each data point. WDYT?

Reply:
I checked that for `TestMySqlTypeMapping.testBasicTypes` failing data points are for double or float. 
Changing the test values from 123.45 to 123.25 solves the problem.

Reply:
> > TestMySqlTypeMapping.testBasicTypes:107 » NoSuchElement
> 
> That also shows other problem. When the test fails, there is no hint what went wrong.
> We could eg issue separate select for each data point. WDYT?

I implemented it that way originally, but the test took several minutes because it had to issue O(N) queries. Maybe we should add this per-column iteration as a debug stage if the combined query fails for any reason?

Reply:
> I checked that for `TestMySqlTypeMapping.testBasicTypes` failing data points are for double or float.
> Changing the test values from 123.45 to 123.25 solves the problem.

Do you think this related to the fact that 123.45 is not exactly representable in floating point?

Reply:
> Maybe we should add this per-column iteration as a debug stage if the combined query fails for any reason?

clever.